test/integration/connections-survive-step-down/connections_survive_step_down.prose.test.ts

```diff
@@ -90,65 +90,57 @@ describe('Connections Survive Primary Step Down - prose', function () {
+    // This test requires a replica set with server version 4.2 or higher.
 
-      // - Set the following fail point: ``{configureFailPoint: "failCommand", mode: {times: 1}, data: {failCommands: ["insert"], errorCode: 10107}}``
-      const failPoint: FailPoint = {
-        configureFailPoint: 'failCommand',
-        mode: { times: 1 },
-        data: { failCommands: ['insert'], errorCode: 10107 }
-      };
+    // - Set the following fail point: ``{configureFailPoint: "failCommand", mode: {times: 1}, data: {failCommands: ["insert"], errorCode: 10107}}``
+    const failPoint: FailPoint = {
+      configureFailPoint: 'failCommand',
+      mode: { times: 1 },
+      data: { failCommands: ['insert'], errorCode: 10107 }
+    };
 
-      it('survives after primary step down', async () => {
-        await client.db('admin').command(failPoint);
-        // - Execute an insert into the test collection of a ``{test: 1}`` document.
-        const error = await collection.insertOne({ test: 1 }).catch(error => error);
-        // - Verify that the insert failed with an operation failure with 10107 code.
-        expect(error).to.be.instanceOf(MongoServerError).and.has.property('code', 10107);
-        // - Execute an insert into the test collection of a ``{test: 1}`` document and verify that it succeeds.
-        await collection.insertOne({ test: 1 });
-        // - If the driver implements the `CMAP`_ specification, verify that no new `PoolClearedEvent`_ has been
-        //   published. Otherwise verify that `connections.totalCreated`_ in `serverStatus`_ has not changed.
-        expect(poolClearedEvents).to.be.empty;
-      });
-    }
-  );
+    it('survives after primary step down', async () => {
+      await client.db('admin').command(failPoint);
+      // - Execute an insert into the test collection of a ``{test: 1}`` document.
+      const error = await collection.insertOne({ test: 1 }).catch(error => error);
+      // - Verify that the insert failed with an operation failure with 10107 code.
+      expect(error).to.be.instanceOf(MongoServerError).and.has.property('code', 10107);
+      // - Execute an insert into the test collection of a ``{test: 1}`` document and verify that it succeeds.
+      await collection.insertOne({ test: 1 });
+      // - If the driver implements the `CMAP`_ specification, verify that no new `PoolClearedEvent`_ has been
+      //   published. Otherwise verify that `connections.totalCreated`_ in `serverStatus`_ has not changed.
+      expect(poolClearedEvents).to.be.empty;
+    });
+  });
 
-  context(
-    'Not Primary - Reset Connection Pool',
-    { requires: { mongodb: '>=4.0.0 <4.2.0', topology: ['replicaset'] } },
-    () => {
-      // This test requires a replica set with server version 4.0.
+  context('Not Primary - Reset Connection Pool', { requires: { topology: ['replicaset'] } }, () => {
+    // This test requires a replica set with server version 4.0.
```

Comment:
Should this test be removed?

Reply:
Actually I left this test as it's still valid that a not primary error resets the connection pool.